File: frontend/src/types/patient.ts
export interface Patient {
  department_name: string;
  id: number;
  full_name: string;
  birth_date?: string;
  gender?: string;
  medical_record_number?: string;
  admission_date: string;
  discharge_date?: string;
  status: 'active' | 'discharged';
  bed_id?: number;
  created_by?: number;
  external_id?: string; // UIID из 1С
}

export interface Bed {
  id: number;
  number: number;
  room_id: number;
  is_occupied: boolean;
  external_id?: string;
  patient?: Patient;
}

export interface Room {
  id: number;
  number: string;
  name?: string;
  description?: string;
  max_beds: number;
  floor?: number;
  wing?: string;
  beds: Bed[];
}

export interface PatientSelectResponse {
  message: string;
  success: boolean;
  welcome_message?: string;
  patient?: Patient;
}

export interface PatientStatus {
  id: number;
  name: string;
  color: string;
}

File: backend/app/schemas/patient.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class Patient(BaseModel):
    id: int
    full_name: str
    birth_date: Optional[datetime] = None
    gender: Optional[str] = None
    medical_record_number: Optional[str] = None
    admission_date: datetime
    discharge_date: Optional[datetime] = None
    status: str
    bed_id: Optional[int] = None
    created_by: Optional[int] = None
    external_id: Optional[str] = None

    # Поля из 1С
    document_id: Optional[str] = None
    branch_id: Optional[str] = None
    department_id: Optional[str] = None
    department_name: Optional[str] = None

    class Config:
        from_attributes = True